spec/unit/dry/cli/cli_spec.rb

```diff
@@ -102,7 +102,7 @@
     end
 
     it 'with option_with_default alias' do
-      output = capture_output { cli.call(arguments: %w[first_arg --option_with_default=test3]) }
+      output = capture_output { cli.call(arguments: %w[first_arg --option-with-default=test3]) }
```

Comment:
how is that (and the rest) related to adding backports for ruby 2.3?

Reply:
https://github.com/ruby/ruby/compare/ruby_2_3...ruby_2_4#diff-e5df590ab3efa95fd6b703249c0e87c8R1389
After ruby 2.4 you able to pass options underscored and it will be dasherized.
I passed it underscored because I called my option underscored, but according to 2.3 rules, I have to pass it dasherized (which were fixed in 2.4). And for anybody, who uses dry-cli with ruby 2.3 won't have to fix their codebase.

Reply:
I've added specs to check previous behaviour for ruby >= 2.4

Reply:
https://github.com/ruby/ruby/blob/ruby_2_4/lib/optparse.rb#L1389
because this line appeared (and 3 more lines with similar functionality)

Reply:
ohhhh I see! thank your this insight